dev_docs/key_concepts/api_authorization.mdx

```diff
@@ -0,0 +1,306 @@
+      },
+    },
+  }, handlerV2)
+  .addVersion({
+    version: '3',
+    validate: false,
+    security: {
+      authz: {
+        requiredPrivileges: ['<privilege_3>'],
+      },
+    },
+  }, handlerV3);
+```
+
+## Authorization Response Available in Route Handlers
+The `AuthzResult` object is available in route handlers, which provides information about the authorization response.
+For example, you have a route that requires  `privilege_3` and ANY of the privileges `privilege_1` OR `privilege_2`:
+```ts
+router.get({
+  path: '/api/path',
+  security: {
+    authz: {
+      requiredPrivileges: ['privilege_3', { anyRequired: ['<privilege_1>', '<privilege_2>'] }],
+    },
+  },
+  ...
+}, (context, request, response) => {
+  // The authorization response is available in `request.authzResult`
+  // {
+  //   "privilege_3": true,
+  //   "privilege_1": true,
+  //   "privilege_2": false
+  // }
+});
+```
+
+## OpenAPI Specification (OAS) documentation
+Based on the security configuration defined in routes, OAS documentation will automatically generate and include description about the required privileges. 
+This makes it easy to view the security requirements of each endpoint in a standardized format, facilitating better understanding and usage by developers or teams consuming the API
+
+To check the OAS documentation for a specific API route and see its security details, you can use the following command:
+```sh
+GET /api/oas?pathStartsWith=/your/api/path
+```
+
+## Migrating from `access` Tags to `security` Configuration
+We aim to use the same privileges that are currently defined with tags `access:<privilege_name>`.
+To assist with this migration, we have created eslint rule `no_deprecated_authz_config`, that will automatically convert your `access` tags to the new `security` configuration.
+It scans route definitions and converts `access` tags to the new `requiredPriviliges` configuration.
+
+Note: The rule is disabled by default to avoid automatic migrations without an oversight. You can perform migrations by running:
+
+**Migrate Routes with Defined Authorization**
+```sh
+MIGRATE_DISABLED_AUTHZ=false npx eslint --ext .ts --fix --rule "@kbn/eslint/no_deprecated_authz_config:error" path/to/your/folder
+```
+We encourage you to migrate routes that are opted out from authorization to new config and provide legitimate reason for disabled authorization.
+It is better to migrate routes opted out from authorization iteratively and elaborate on the reasoning. 
+Probably, you can consider enabling authorization for these routes in the future.
+
+**Migrate All Routes (Including Those Opted Out from Authorization)**
+```sh
+MIGRATE_DISABLED_AUTHZ=true npx eslint --ext .ts --fix --rule "@kbn/eslint/no_deprecated_authz_config:error" path/to/your/folder
+```
+
+## Questions?
+If you have any questions or need help with API authorization, please reach out to the Kibana Security team on the `#kibana-security` Slack channel.
```

Comment:
These docs are in the public repo, so it may be best to instead mention our GH team.
```suggestion
If you have any questions or need help with API authorization, please reach out to the `@elastic/kibana-security` team.
```